### DISCUSSION TITLE
v0.5.0 - Difference between "subscriptionExpireTime" & "expiresAt" in "SubscriptionInfo"

### DISCUSSION THREAD

**Discussion by maxl2287** [camaraproject/DeviceStatus#105] [Q&A] [answered]
I just want to know what the difference is between these values?
I hope somebody can help me here.

**Comment by bigludo7** [comment, marked as answer]
Hello @maxl2287 
Looking in the guidelines:

subscriptionExpireTime: Date when the event subscription should end. Provided by API requester. The Server may reject the subscription if the period requested do not comply with Telco Operator policies (i.e. to avoid unlimited time subscriptions). In this case server returns exception 403 "SUBSCRIPTION_PERIOD_NOT_ALLOWED"

expiresAt: Date when the event subscription will expire. This attribute must not be present in the POST request as it is provided by API server.

Tricky but I saw 2 UC: 

- subscriptionExpireTime is not mandatory  in the request. Suppose that Telco Operator policies is to limit subscription time (max 6 months for example). It allows to value the expiresAt at future date when automatically the subscription will be terminated. 
- If subscriptionExpireTime  is provided but too far and Telco Operator policies does not want to use 403 (we did not enforce this rule) then expiresAt could be helpful.

**Comment by akoshunyadi** [comment]
Yes, I think the simplest case is when subscriptionExpireTime is missing in the request, then only expiresAt contains the expiry time. 
The names can be in deed confusing, but to change them you have to go to commonalities. 🙂

**Comment by akoshunyadi** [comment]
Issue solved and can be closed.
